### PR TITLE
New model: Labor supply on intensive margin

### PR DESCRIPTION
**Pull request by mnwhite**
This PR provides a new model for HARK, in which agents make both a consumption-saving decisions and a (continuous) labor-leisure decision.  Agents observe their permanent and transitory productivity shocks before deciding how much labor to supply each period, so TranShk is a second state variable.  Utility function has a Cobb-Douglas-like aggregator for leisure and consumption under a single CRRA term.

Most model code was written by Tiphanie Magne in 2017, with some cleanup by me in 2019 and then a bit more by Jack Shiqi Li.

I could swear this was merged into master months ago, but apparently not.

### CODE REVIEW

**Comment by sbenthall** [comment]
What if, for new Models, rather than putting all the test and demo code into the `__main__` method in the model class file, rather:
 - the demo code was put into `examples/` as a notebook as per #455 
 - the test code had automated tests that would be run with Travis: https://github.com/econ-ark/HARK/tree/master/HARK/tests

This has two benefits:
 - the demonstration code would then be available to be ingested into ReadTheDocs
 - the test code would serve as further confirmation of the robustness of the more basic functionality, which would make the library more robust to refactoring, etc.

**Comment by mnwhite** [comment]
Yes, that's a better structure.  I'll move the code.

**Comment by mnwhite** [comment]
Ok, the main block code has been moved to a new examples file, and tested in place.  This revealed a bunch of problems with the code (now fixed), so it was a very good idea.

File: HARK/ConsumptionSaving/ConsLaborModel.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+        Returns
+        -------
+        None
+        '''
+        raise NotImplementedError()
+
+        
+    def getControls(self):
+        '''
+        Calculates consumption for each consumer of this type using the consumption functions.
```
Comment (sbenthall) [inline HARK/ConsumptionSaving/ConsLaborModel.py]: I think this docstring is incorrect. There are more controls than just consumption in this model

**Comment by MridulS** [comment]
for the unit tests in https://github.com/econ-ark/HARK/pull/455 I use ideas from the example and then create incremental unit test. Example: testing one function `checkMarkovInputs` of `MarkovConsumerType` https://github.com/econ-ark/HARK/blob/2b99308daabea1dc4c6109509793ddfbbb269068/HARK/ConsumptionSaving/tests/test_ConsMarkovModel.py

**Comment by mnwhite** [comment]
I fixed that docstring error just now (thanks).  I've never made automated tests before; I'll try to get around to figuring out how to do that and put them in here.

**Comment by sbenthall** [comment]
@mnwhite I'd be happy to do a quick call with you some time about automated tests!
we could even pair code it (remote) if you had the time for it.

**Comment by sbenthall** [comment]
Shall I merge this in? It's better if there are tests, but there is no such general policy requiring tests on PRs at this point, as far as I know.
If this feature is needed downstream, might as well merge it.
My offer to walk you through automated testing stands, any time you've got time.

**Comment by mnwhite** [comment]
Thank you!  We will make tests, I swear.

**Pull request merged**